contrib/lkt/language/parser.py

```diff
@@ -332,42 +350,15 @@ def type_mismatch_error(expected=T.TypeDecl.entity, got=T.TypeDecl.entity):
             S("`").concat(got.full_name).concat(S("`"))
         )
 
-    @langkit_property(return_type=T.TreeSemanticResult, public=True)
-    def check_semantic():
-        """
-        Run name and type resolution on all relevant entities in this subtree,
-        aggregating and returning the results.
-
-        If you ran this and the result's `has_error` flag is false, then you
-        *know* that you can call semantic properties such as `referenced_decl`
-        and `expr_type`, and those will never raise.
-        """
-        res = Var(Entity.check_sem_internal())
-
-        # Crude sanity check: since for the moment the Try blocks in
-        # check_sem_internal will catch any exception happening in subcalls to
-        # semantic properties, we want to check that, if we have internal
-        # errors, we also have user errors (i.e. diagnostics). If we have an
-        # internal error and no user errors, then we caught something we were
-        # not supposed to.
-        #
-        # TODO: Assess whether this is still needed when we migrate to a better
-        # error handling mechanism (see check_semantic_impl).
-
-        errs = Var(res.results.filter(lambda r: r.has_error))
-        diags = Var(errs.filter(lambda r: Not(r.error_message.is_null)))
-
-        has_internal_errors = Var(errs.any(lambda r: r.error_message.is_null))
-
-        return If(
-            has_internal_errors & (diags.length == 0),
-            PropertyError(T.TreeSemanticResult,
-                          "ERROR: internal errors without diagnostics"),
-            res
+    @langkit_property(return_type=T.SemanticResult.array)
+    def check_legality_impl():
```

Comment:
Can you add a docstring for this? `check_legality` (at least defined line 403) does not call this, which is the usual pattern with `_impl` properties (here it’s the other way around)